ui/imports/shared/views/chat/ImageContextMenu.qml

```diff
@@ -1,29 +1,38 @@
+import QtQuick 2.15
 import StatusQ.Popups 0.1
-
 import utils 1.0
 
 StatusMenu {
     id: root
 
+    property string url
     property string imageSource
 
-    readonly property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
+    QtObject {
+        id: d
+        property bool isUnfurled: (!!url&&url!=="")
+        readonly property bool isGif: root.imageSource.toLowerCase().endsWith(".gif")
+    }
 
     StatusAction {
-        text: root.isGif ? qsTr("Copy GIF") : qsTr("Copy image")
+        text: qsTr("Copy ") + (d.isUnfurled ? qsTr("link") :
+                              d.isGif ? qsTr("GIF") : qsTr("image"))
         icon.name: "copy"
         enabled: !!root.imageSource
         onTriggered: {
+            d.isUnfurled ? Utils.copyToClipboard(url) :
             Utils.copyImageToClipboardByUrl(root.imageSource)
         }
     }
 
     StatusAction {
-        text: root.isGif ? qsTr("Download GIF") : qsTr("Download image")
+        text: d.isUnfurled ? qsTr("Open link") : (qsTr("Download ")
+              + (d.isGif ? qsTr("GIF") : qsTr("image")))
         icon.name: "download"
```

Comment:
There's also a different icon for `Open link`.

https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=22376-63726&mode=design&t=D9uNl4Wn7WX8tNvR-4

![image](https://github.com/status-im/status-desktop/assets/25482501/6279bfff-368d-4376-8377-77d5c056ddbf)

Reply:
Considering this, do you think it would be cleaner to have 2 separate `StatusAction`s for link case here? Then it would also be easy later to show all 4 options for unfurled images if we decide to.
Multiple conditions like `d.isUnfurled ? ... : d.isGIf ? ... : ...` doesn't really look good 😄

Reply:
@igor-sirotin icon updated. regarding states, I think Menu since it's a QtObject doesn't support states.

Reply:
I didn't mean to use states. I suggested to use separate `StatusAction`s. 
Like this:

```qml
StatusAction {
    visible: !d.isUnfurled
    text: d.isGif ? "Copy GIF" : "Copy image"
}
StatusAction {
    visible: d.isUnfurled
    text: "Copy link"
}
```

And same for download/open.

Reply:
I didn't mean to use states. I suggested to use separate `StatusAction`s. 
Like this:

```qml
StatusAction {
    visible: !d.isUnfurled
    text: d.isGif ? "Copy GIF" : "Copy image"
}
StatusAction {
    visible: d.isUnfurled
    text: "Copy link"
}
```

And same for download/open.

This would be much more readable and easier to maintain.
And, as I said, if we decide to leave all 4 options for unfurled images, it will be just a question of removing `visible` binding.

Reply:
Agreed, that's what I suggested too

Reply:
@igor-sirotin @caybro same here, [Action](https://doc.qt.io/qt-5/qml-qtquick-controls2-action.html) is a QtObject, hence no visible property there

Reply:
We use `enabled` for that purpose

Reply:
pushed